### PR TITLE
Add serialization and blockstore stats for common tests

### PR DESCRIPTION
**Pull request by austinabell**
- Adds serialization vectors and blockstore stats to the tests that are replicated

The stats include reads, writes, bytes read, bytes written to use to compare against since Filecoin uses reads, writes, and bytes written for gas usage.

Rust equivalent test additions: https://github.com/ChainSafe/forest/pull/716

There are a few cases where there are many more writes than necessary, but I will open up an issue to explain

### CODE REVIEW

File: amt_test.go [hunk 1]
```diff
@@ -726,6 +815,60 @@ func TestForEach(t *testing.T) {
...
+				return err
+			}
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, "bafy2bzaced44wtasbcukqjqicvxzcyn5up6sorr5khzbdkl6zjeo736f377ew", c.String())
+	assert.Equal(t, bsStats{r: 12, w: 12, br: 573, bw: 573}, trackingBs.stats)
```
Comment (rvagg) [inline amt_test.go:871]: these stats 😱 are why I think we need to work on a collapsable form https://github.com/filecoin-project/go-amt-ipld/issues/17#issuecomment-668568876 .. I don't have any idea how common this kind of perverse sparseness is in actual Filecoin data though

**Comment by rvagg** [comment]
these lgtm but needs a rebase and to have the commented code dealt with - can `TestMain` come back? and `TestRoundTrip` I think has been replaced

> There are a few cases where there are many more writes than necessary, but I will open up an issue to explain

would be very interested to hear more on this if you still have these top of mind @austinabell

File: amt_test.go (hunk 1 above)
Comment (austinabell) [reply amt_test.go:871]: the only stat that isn't used in Filecoin was bytes read, all other 3 are relevant

**Comment by austinabell** [comment]
> these lgtm but needs a rebase and to have the commented code dealt with - can `TestMain` come back? and `TestRoundTrip` I think has been replaced
> 
> > There are a few cases where there are many more writes than necessary, but I will open up an issue to explain
> 
> would be very interested to hear more on this if you still have these top of mind @austinabell

Sorry, I don't have the bandwidth to do this anytime soon, the motivation was to make sure blockstore usages for the amt is consistent for all cases as there were some strange patterns, but I don't know the current status or if this would be useful for other impls now that this is probably stabilized. Would be good to make sure things don't change in any case, but I don't have an opinion or motivation to push for this myself so feel free to close or modify this PR.

**Comment by austinabell** [comment]
closing because stale

**Pull request closed without merging**